File: zy70049/src/routes/tasks.js
const express = require('express');
const router = express.Router();
const TaskService = require('../services/taskService');

router.post('/generate', (req, res) => {
  try {
    const result = TaskService.generateExpiryTasks(req.body.targetDate);
    res.json(result);
  } catch (err) {
    res.status(500).json({ error: err.message });
  }
});

router.get('/', (req, res) => {
  const tasks = TaskService.getTasks(req.query);
  res.json(tasks);
});

router.get('/:id', (req, res) => {
  const task = TaskService.getTask(req.params.id);
  if (!task) return res.status(404).json({ error: '任务不存在' });
  res.json(task);
});

router.post('/:id/extend', (req, res) => {
  try {
    const { extension_days, reason, operator } = req.body;
    const task = TaskService.extendTask(req.params.id, extension_days, reason, operator);
    res.json(task);
  } catch (err) {
    res.status(400).json({ error: err.message });
  }
});

router.get('/jobs/runs', (req, res) => {
  const runs = TaskService.getJobRuns(req.query.job_name || 'generate_expiry_tasks');
  res.json(runs);
});

module.exports = router;

File: xy11275/forklift_charging_system/app/api/tasks.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from typing import List, Optional
from datetime import datetime
from app.database import get_db
from app.schemas import ChargingTaskCreate, ChargingTask as ChargingTaskSchema, BatchResult
from app.services.task_service import TaskService

router = APIRouter(prefix="/tasks", tags=["充电任务"])


@router.post("/", summary="创建单个充电任务")
def create_task(
    task: ChargingTaskCreate,
    operator: str = Query(..., description="操作人"),
    db: Session = Depends(get_db)
):
    service = TaskService(db)
    db_task, error = service.create_task(task, operator)
    
    if error and db_task.status == "rejected":
        return {
            "task": db_task,
            "status": "rejected",
            "reason": error
        }
    
    return {
        "task": db_task,
        "status": "approved"
    }


@router.post("/batch", response_model=BatchResult, summary="批量创建充电任务")
def batch_create_tasks(
    tasks: List[ChargingTaskCreate],
    operator: str = Query(..., description="操作人"),
    db: Session = Depends(get_db)
):
    service = TaskService(db)
    result = service.batch_create_tasks(tasks, operator)
    return result


@router.get("/", response_model=List[ChargingTaskSchema], summary="获取任务列表")
def get_tasks(
    requested_by: Optional[str] = Query(None, description="按负责人筛选"),
    status: Optional[str] = Query(None, description="按状态筛选"),
    shift: Optional[str] = Query(None, description="按班次筛选"),
    skip: int = 0,
    limit: int = 100,
    db: Session = Depends(get_db)
):
    service = TaskService(db)
    return service.get_tasks(
        requested_by=requested_by,
        status=status,
        shift=shift,
        skip=skip,
        limit=limit
    )


@router.get("/{task_id}", response_model=ChargingTaskSchema, summary="获取单个任务详情")
def get_task(task_id: int, db: Session = Depends(get_db)):
    service = TaskService(db)
    task = service.get_task_by_id(task_id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")
    return task


@router.get("/{task_id}/reason", summary="获取任务拦截/放行原因")
def get_task_reason(task_id: int, db: Session = Depends(get_db)):
    service = TaskService(db)
    task = service.get_task_by_id(task_id)
    if not task:
        raise HTTPException(status_code=404, detail="任务不存在")
    
    return {
        "task_id": task_id,
        "status": task.status,
        "reason": task.reason
    }
